src/libraries/System.Net.Http/src/System/Net/Http/MetricsHandler.cs

```diff
@@ -0,0 +1,252 @@
+                tags.Add("port", requestUri.Port);
+            }
+        }
+        tags.Add("method", request.Method.Method);
+
+        return tags;
+    }
+
+    void IHttpMetricsLogger.LogRequestFailed(HttpResponseMessage response, Exception exception)
+    {
+        Debug.Assert(response.RequestMessage is not null);
+        TagList tags = InitializeCommonTags(response.RequestMessage);
+        ApplyExtendedTags(ref tags, response.RequestMessage, response, exception);
+        _failedRequests.Add(1, tags);
+    }
+
+    private static void CreateInstruments(Meter meter, out UpDownCounter<long> currentRequests, out Counter<long> failedRequests, out Histogram<double> requestsDuration)
+    {
+        currentRequests = meter.CreateUpDownCounter<long>(
+            "http-client-current-requests",
+            description: "Number of outbound HTTP requests that are currently active on the client.");
+        failedRequests = meter.CreateCounter<long>(
+            "http-client-failed-requests",
+            description: "Number of outbound HTTP requests that have failed.");
+        requestsDuration = meter.CreateHistogram<double>(
+            "http-client-request-duration",
+            unit: "s",
+            description: "The duration of outbound HTTP requests.");
+    }
+
+    private static class StatusCodeCache
+    {
+        private static readonly object OK = (int)HttpStatusCode.OK;
+        private static readonly object Created = (int)HttpStatusCode.Created;
+        private static readonly object Accepted = (int)HttpStatusCode.Accepted;
+        private static readonly object NoContent = (int)HttpStatusCode.NoContent;
+        private static readonly object Moved = (int)HttpStatusCode.Moved;
+        private static readonly object Redirect = (int)HttpStatusCode.Redirect;
+        private static readonly object NotModified = (int)HttpStatusCode.NotModified;
+        private static readonly object InternalServerError = (int)HttpStatusCode.InternalServerError;
+
+        public static object GetBoxedStatusCode(HttpStatusCode statusCode) => statusCode switch
+        {
+            HttpStatusCode.OK => OK,
+            HttpStatusCode.Created => Created,
+            HttpStatusCode.Accepted => Accepted,
+            HttpStatusCode.NoContent => NoContent,
+            HttpStatusCode.Moved => Moved,
+            HttpStatusCode.Redirect => Redirect,
+            HttpStatusCode.NotModified => NotModified,
+            HttpStatusCode.InternalServerError => InternalServerError,
+            _ => (int)statusCode
+        };
+    }
+
+    private sealed class SharedMeter : Meter
+    {
+        internal UpDownCounter<long> _currentRequests;
+        internal Counter<long> _failedRequests;
+        internal Histogram<double> _requestsDuration;
+
+        public SharedMeter()
+            : base("System.Net.Http")
+        {
+            CreateInstruments(this, out _currentRequests, out _failedRequests, out _requestsDuration);
+        }
```

Comment:
This isn't necessary. The create methods on Meter, e.g. `Meter.CreateUpDownCounter`, have an internal cache. They check if the counter with that name already exists and returns the cached copy.

See https://github.com/dotnet/runtime/blob/5ff4fdd0250e73b34cecb3c994746bcb9d7325cb/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Meter.cs#L474